File: nodes/get-object-property/get-object-property.process.py
"""
Process function for the Get Object Property node.
Extracts a specific property value from an object by key name.
"""
from typing import Any, Dict

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    object_input = inputs.get("object")
    key = inputs.get("key")
    default_value = settings.get("default_value")
    
    # Validate inputs
    if not isinstance(object_input, dict):
        return {
            "value": default_value,
            "exists": False,
            "key": key or ""
        }
    
    if not isinstance(key, str):
        return {
            "value": default_value,
            "exists": False,
            "key": key or ""
        }
    
    # Check if property exists
    exists = key in object_input
    value = object_input.get(key, default_value) if exists else default_value
    
    return {
        "value": value,
        "exists": exists,
        "key": key
    }

File: sdks/nodejs/nodes/get-object-property/get-object-property.process.js
/**
 * Process function for the Get Object Property node.
 * Extracts a specific property value from an object by key name.
 */
export default async ({inputs, settings, config}) => {
  const { object, key } = inputs;
  const { default_value } = settings;
  
  // Validate inputs
  if (!object || typeof object !== 'object') {
    return {
      value: default_value,
      exists: false,
      key: key || ''
    };
  }
  
  if (!key || typeof key !== 'string') {
    return {
      value: default_value,
      exists: false,
      key: key || ''
    };
  }
  
  // Check if property exists
  const exists = key in object;
  const value = exists ? object[key] : default_value;
  
  return {
    value: value,
    exists: exists,
    key: key
  };
};
